pymc3/distributions/continuous.py

```diff
@@ -2780,46 +2789,28 @@ class HalfStudentT(PositiveContinuous):
         with pm.Model():
             x = pm.HalfStudentT('x', lam=4, nu=10)
     """
+    rv_op = halfstudentt
+
+    @classmethod
+    def dist(cls, nu=1, sigma=None, lam=None, sd=None, *args, **kwargs):
 
-    def __init__(self, nu=1, sigma=None, lam=None, sd=None, *args, **kwargs):
-        super().__init__(*args, **kwargs)
         if sd is not None:
             sigma = sd
 
-        self.mode = at.as_tensor_variable(0)
         lam, sigma = get_tau_sigma(lam, sigma)
-        self.median = at.as_tensor_variable(sigma)
-        self.sigma = self.sd = at.as_tensor_variable(sigma)
-        self.lam = at.as_tensor_variable(lam)
-        self.nu = nu = at.as_tensor_variable(floatX(nu))
+        mode = at.as_tensor_variable(0)
+        median = at.as_tensor_variable(sigma)
+        sigma = sd = at.as_tensor_variable(sigma)
+        lam = at.as_tensor_variable(lam)
+        nu = nu = at.as_tensor_variable(floatX(nu))
 
         assert_negative_support(sigma, "sigma", "HalfStudentT")
         assert_negative_support(lam, "lam", "HalfStudentT")
         assert_negative_support(nu, "nu", "HalfStudentT")
 
-    def random(self, point=None, size=None):
-        """
-        Draw random values from HalfStudentT distribution.
-
-        Parameters
-        ----------
-        point: dict, optional
-            Dict of variable values on which random values are to be
-            conditioned (uses default point if not specified).
-        size: int, optional
-            Desired size of random sample (returns one sample if not
-            specified).
-
-        Returns
-        -------
-        array
-        """
-        # nu, sigma = draw_values([self.nu, self.sigma], point=point, size=size)
-        # return np.abs(
-        #     generate_samples(stats.t.rvs, nu, loc=0, scale=sigma, dist_shape=self.shape, size=size)
-        # )
+        return super().dist([nu, sigma, lam, sd], *args, **kwargs)
```

Comment:
Only one default parametrization should go into `super().dist`. You can check the already refactored t-student for an example.